src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderLabel.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Security.Cryptography.Cose
+{
+    [DebuggerDisplay("Label = {LabelAsInt32.HasValue ? LabelAsInt32 : LabelAsString}, Type = {LabelAsInt32.HasValue ? typeof(int) : typeof(string),nq}")]
```

Comment:
It's better in general to keep code out of DebuggerDisplay.  The debugger needs to evaluate it, and this will likely fail if a project is using something other than C#.  Better to just have a private property like `private string DebuggerLabel => ...;` that computes the right thing, and then here have "Label = {DebuggerLabel}".